### PR TITLE
Development

### PR DESCRIPTION
**Pull request by GHTeodor**

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -1,21 +1,52 @@
...
-var warrior1 = Menu.Hero(1);
-var warrior2 = Menu.Hero(2);
+do
+{
+    List<Warrior> warriors = new List<Warrior>();
 
-int addHero1 = SuperPowerGenerator.GetRandom();
-Console.WriteLine("\n Що ви хочете покращити першому Герою №1 на [" + addHero1 + "] пунктів");
-Menu.Update(warrior1, addHero1);
+    int iter = 1;
```
Comment (YevhenVolynets) [inline Program.cs:11]: why do you need that field?

File: Program.cs [hunk 2]
```diff
@@ -1,21 +1,52 @@
...
-int addHero2 = SuperPowerGenerator.GetRandom();
-Console.WriteLine("\n Що ви хочете покращити першому Герою №2 на [" + addHero2 + "] пунктів");
-Menu.Update(warrior2, addHero2);
+    while (warriors.Count() < players)
+    {
+        warriors.Add(Menu.Hero(warriors.Count() + 1));
+        
+        int addToHero = SuperPowerGenerator.GetRandom();
+        Console.WriteLine($"\n Що ви хочете покращити Герою №{iter--} на [{addToHero}] пунктів");
+        Menu.Update(warriors.Skip(iter++).First(), addToHero);
```
Comment (YevhenVolynets) [inline Program.cs:20]: It is not a best solution to get hero. You can assign hero to local field (line 16) and use that field

**Comment by YevhenVolynets** [comment]
image
Why do you need break?

**Comment by YevhenVolynets** [comment]
image
Why do you need return type? What if you'll try to update 2 or more fields?

**Comment by YevhenVolynets** [comment]
image
Random is not allowed for that class

**Comment by YevhenVolynets** [comment]
image
Properties should be in upper case. Also if property has only get, it is used " => " style.

**Comment by YevhenVolynets** [comment]
image
The name of the method should indicate its essence. Should be GetNumberOfPlayers or something like that

**Comment by YevhenVolynets** [comment]
The same - NewGameAgain class. Also class names should indicate the logic for which they were created. NewGameAgain => GameHelper or StartGameHelper
NumberOfPlayers => PlayerHelper, GameGenerator....

**Comment by YevhenVolynets** [comment]
image
It is not the best solution for grouping. Just imagine, you'll need to show all heroes at the end of the last game. You won't do it because the list will be empty.

**Comment by YevhenVolynets** [comment]
image
Scene class shouldn't be responsible for Attack and Defense. What you'll do if you need to update Defense for only one specific warrior?

**Review by YevhenVolynets — CHANGES_REQUESTED** [review]

**Review by YevhenVolynets — CHANGES_REQUESTED** [review]
Added all comments

File: Program.cs (hunk 1 above)
Comment (GHTeodor) [reply Program.cs:11]: Just to show number of hero in Console (like in the next remark, "Hero №{iter}")

Console.WriteLine($"\n What do you want to increase Hero №{iter--} by [{addToHero}] points");

**Pull request merged**